templates/pkg/resource/sdk.go.tpl

```diff
@@ -212,7 +212,7 @@ func (rm *resourceManager) updateConditions (
 		}
 	}
 
-	if rm.terminalAWSError(err) || err ==  ackerr.SecretTypeNotSupported || err == ackerr.SecretNotFound {
+	if !r.IsBeingDeleted() && (rm.terminalAWSError(err) || err ==  ackerr.SecretTypeNotSupported || err == ackerr.SecretNotFound) {
```

Comment:
I suppose this is fine. But, would it be safer to just let the Terminal Condition be set on the resource even if it's being deleted?

Reply:
I don't think it's safe to set Terminal Condition here because if deletion encounters an error, then wouldn't the controller stop reconciling without deleting the finalizer?

Reply:
will this as-is for now, until we have another strategy. If a customer hits this(should be a corner case), the worse case is controller retires with back off

Reply:
> I don't think it's safe to set Terminal Condition here because if deletion encounters an error, then wouldn't the controller stop reconciling without deleting the finalizer?

Yes, and that is what the controller should do I think. If the delete operation failed on the backend AWS service API call, then the resource will not be deleted from the AWS service, so we should keep the resource around on the k8s side, but in show a terminal condition.

Reply:
> Yes, and that is what the controller should do I think. If the delete operation failed on the backend AWS service API call, then the resource will not be deleted from the AWS service, so we should keep the resource around on the k8s side, but in show a terminal condition.

Does this mean the only way to delete that resource (after resolving any errors) is for users to manually remove the finalizer via `kubectl edit` ? That seems weird cx or am I missing some details

Reply:
> > Yes, and that is what the controller should do I think. If the delete operation failed on the backend AWS service API call, then the resource will not be deleted from the AWS service, so we should keep the resource around on the k8s side, but in show a terminal condition.
> 
> Does this mean the only way to delete that resource (after resolving any errors) is for users to manually remove the finalizer via `kubectl edit` ? That seems weird cx or am I missing some details

The correct user experience for manually removing a resource that is in a stuck state: using `kubectl delete --force`